sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/AvoidNotShortCircuitOperatorsForBooleanCheck.java

```diff
@@ -160,14 +151,12 @@ public final static boolean isBooleanType(final DetailAST node)
      */
     public final boolean isBooleanExpression(final DetailAST node)
     {
-
         DetailAST curNode = node;
 
         final List<String> childNames = getSupportedOperandsNames(curNode);
         final List<String> booleanVariablesNames = new LinkedList<String>();
 
-        while (curNode != null
```

Comment:
Can you meet `null` in hierarchy between `EXPR` and any of those values? I Believe not and that's why I dropped it.